src/nisarqa/raster_classes.py

```diff
@@ -626,34 +656,127 @@ def decimate_raster_array_to_square_pixels(
+    raster_obj: RadarRaster | GeoRaster,
+) -> tuple[np.ndarray, int, int]:
+    """
+    Decimate *Raster's data array to square pixels and also return strides.
+
+    Pixels will be approx. square in the X and Y direction.
 
     Parameters
     ----------
     raster_obj : RadarRaster or GeoRaster
         *Raster object whose .data attribute will be read into memory
-        and decimated along the first two dimensions to square pixels.
+        and decimated along the first two dimensions to approx. square pixels.
 
     Returns
     -------
     out : numpy.ndarray
         Copy of raster_obj.data array that has been decimated along the
-        first two dimensions to have square pixels.
+        first two dimensions to have approx. square pixels.
+    ky, kx : int
+        The stride used for performing decimation in the X and Y directions,
+        respectively.
+    """
+    # Decimate to square pixels.
+    return decimate_array_to_square_pixels_with_strides(
+        arr=raster_obj.data[...],
+        y_axis_spacing=raster_obj.y_axis_spacing,
+        x_axis_spacing=raster_obj.x_axis_spacing,
+    )
+
+
+def decimate_array_to_square_pixels(
+    arr: np.ndarray, y_axis_spacing: float, x_axis_spacing: float
+) -> np.ndarray:
+    """
+    Decimate array to approx. square pixels in X and Y direction.
+
+    Parameters
+    ----------
+    arr : np.ndarray
+        Input array to be decimated along the first two dimensions to
+        approx. square pixels.
+    y_axis_spacing : float
+        Pixel Spacing in Y direction (azimuth for range-Doppler rasters).
+    x_axis_spacing : float
+        Pixel Spacing in X direction (range for range-Doppler rasters).
+
+    Returns
+    -------
+    out : numpy.ndarray
+        Copy of `arr` that has been decimated along the first two
+        dimensions to have approx. square pixels.
+    """
+    out, _, _ = decimate_array_to_square_pixels_with_strides(
+        arr=arr, y_axis_spacing=y_axis_spacing, x_axis_spacing=x_axis_spacing
+    )
+
+    return out
+
+
+def decimate_array_to_square_pixels_with_strides(
+    arr: np.ndarray, y_axis_spacing: float, x_axis_spacing: float
+) -> tuple[np.ndarray, int, int]:
+    """
+    Decimate array to approx. square pixels in X and Y direction.
+
+    Parameters
+    ----------
+    arr : np.ndarray
+        Input array to be decimated along the first two dimensions to
+        approx. square pixels.
+    y_axis_spacing : float
+        Pixel Spacing in Y direction (azimuth for range-Doppler rasters).
+    x_axis_spacing : float
+        Pixel Spacing in X direction (range for range-Doppler rasters).
+
+    Returns
+    -------
+    out : numpy.ndarray
+        Copy of `arr` that has been decimated along the first two
+        dimensions to have approx. square pixels.
+    ky, kx : int
+        The stride used for performing decimation in the X and Y directions,
+        respectively.
     """
-    arr = raster_obj.data[...]
+    arr = np.copy(arr)
```

Comment:
Don't think we need to make a copy of the input array since we're not modifying the input array (hopefully...)